src/test/java/stepDefinition/GenericsStepDef.java

```diff
@@ -0,0 +1,38 @@
+package stepDefinition;
+
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import helpers.ScenarioContext;
+import org.openqa.selenium.By;
+import pages.CategoryPage;
+
+import static helpers.ActionHelper.clickOnElement;
+import static helpers.ElementSearchUtils.getElementByName;
+import static helpers.ElementSearchUtils.getPage;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public class GenericsStepDef extends AbstractStepDef {
+
+    @Then("^'(.*)' page is displayed$")
+    public void pageIsDisplayed(String pageName) {
+        assertThat(String.format("%s page is not displayed", pageName),
+                webDriver.getCurrentUrl(),
+                is(getPage(pageName.concat("Page")).getPageUrl()));
+        waitForPageLoaded();
```

Comment:
why waitForPageload is after assert?

Reply:
Because sometimes is not enough time to find an element on a new page.